chain/chain/src/flat_storage_creator.rs

```diff
@@ -403,10 +410,13 @@ pub struct FlatStorageCreator {
 impl FlatStorageCreator {
     /// For each of tracked shards, either creates flat storage if it is already stored on DB,
     /// or starts migration to flat storage which updates DB in background and creates flat storage afterwards.
+    ///
+    /// Also resumes any resharding operation which was already in progress.
     pub fn new(
         epoch_manager: Arc<dyn EpochManagerAdapter>,
         runtime: Arc<dyn RuntimeAdapter>,
         chain_store: &ChainStore,
+        flat_storage_resharder: &FlatStorageResharder,
```

Comment:
I didn't completely understand why is flat_storage_resharder a part of FlatStorageCreator? The FlatStorageCreator is created at the time of calling new() in client. The `Self::create_flat_storage_for_current_epoch` is also thus called only in the beginning right? How do we later set the status of some flat storage to `FlatStorageStatus::Resharding`?

Reply:
Basically the `FlatStorageResharder` is part of client. 

`FlatStorageCreator` uses a reference to the resharder to handle the case when the node restarts and flat storage resharding was in progress. I did like this because the creator already has logic to handle resuming flat storage from the previous state.

In normal flow the resharding of flat storage would be initiated by the resharding manager, but that part is not done yet